test-error/apps/payslip-validator/src/main/java/com/redhat/training/payslipvalidator/processor/AmountProcessor.java

```diff
@@ -0,0 +1,26 @@
+package com.redhat.training.payslipvalidator.processor;
+
+import static org.apache.camel.builder.xml.XPathBuilder.xpath;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.w3c.dom.NodeList;
+
+public class AmountProcessor implements Processor {
+
+	@Override
+	public void process(Exchange exchange) throws Exception {
+		NodeList result = xpath(
+				"/payslip/payslipItems/payslipItem/payslipItemQty/text()"
+		).evaluate(exchange, NodeList.class);
+
+		int total = 0;
+
+		for (int i = 0; i < result.getLength(); i++) {
```

Comment:
Use functional streams, maps instead of for loops? We are using jdk 11 correct?

Reply:
Disclaimer: I'm not a Java expert..

I've been looking into this and `NodeList` is a simple interface that is not iterable and I can't rely in the implementation to iterate through the child nodes. Transforming the `NodeList` object into into something iterable to me is a little bit of ugly and adds a lot of overhead in the code.. but.. what about having something like this?:

```
	@Override
	public void process(Exchange exchange) throws Exception {
		NodeList result = xpath(
				"/payslip/payslipItems/payslipItem/payslipItemQty/text()"
		).evaluate(exchange, NodeList.class);

		Stream<Node> nodeStream = IntStream.range(0, result.getLength())
				.mapToObj(result::item);

		exchange.getIn().setHeader(
				"totalPayslipUnits",
				nodeStream.mapToInt(node -> Integer.parseInt(node.getNodeValue())).sum()
		);
	}
```